solution.cpp

```diff
@@ -0,0 +1,87 @@
+/* ----------------------------------------------------------------------------
+* Copyright & copy; 2015 Matthew Low <mcorelow@csu.fullerton.edu>
+* Released under the [MIT License] (http://opensource.org/licenses/MIT)
+* ------------------------------------------------------------------------- */
+
+//Various functions using recursion
+
+#include <iostream>
+using std::cout;
+using std::endl;
+using std::cin;
+
+//prototypes
+int abs(int);
+int gcd(int,int);
+int gcd_iterative(int, int);
+int fibonacci(int, int a = 1, int b = 1);
+	
+int main(){
+	int gnum1, gnum2, fnum;
+	
+	//gcd 
+	cout << "Enter two ints and to find the gcd: ";
+	cin >> gnum1 >> gnum2;
+	cout << "GCD: " << gcd(gnum1,gnum2) << endl;
+	cout << "GCD iterative: " << gcd_iterative(gnum1,gnum2) << endl;
+	cout << endl;
+	
+	//fibonacci
+	cout << "Enter an int to find the nth fibonacci number: ";
+	cin >> fnum;
+	cout << fibonacci(fnum) << endl;
+return 0;
+}
```

Comment:
probably easier to test if you just hardcode values (or even use random values), rather than using `cin`